beetsplug/web_tagger.py

```diff
@@ -0,0 +1,108 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016, Tigran Kostandyan <t.kostandyan1@gmail.com>
+#
+# Permission is hereby granted, free of charge, to any person obtaining
+# a copy of this software and associated documentation files (the
+# "Software"), to deal in the Software without restriction, including
+# without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so, subject to
+# the following conditions:
+#
+# The above copyright notice and this permission notice shall be
+# included in all copies or substantial portions of the Software.
+
+""" Runs local http server for MusicBrainz web tagger and handles users' requests """
+
+from __future__ import division, absolute_import, print_function
+
+import threading
+import socket
+import webbrowser
+try:
+    import urllib.parse as urlparse
+    from urllib.parse import urlencode
+except ImportError: # python 2 support
+    import urlparse
+    from urlparse import urlparse
+from beets.plugins import BeetsPlugin
+from beets import autotag
+from beets.ui.commands import PromptChoice
+from beets import ui
+
+
+PORT = 8000  # Temporary for PR
+
+
+def parse(data):
+    data = data.splitlines()
+    s = data[0].decode()
+    url = ''
+    for char in s[5:]:
+        if char != ' ':
+            url += char
+        else:
+            break
+    parsed = urlparse.urlparse(url)
+    return str(urlparse.parse_qs(parsed.query)['id'][0])

```

Comment:
Aha, got it! It seems like the right thing to do is to use a real HTTP parser then. Here's one way to do it, if you want to avoid using a full HTTPServer: http://stackoverflow.com/a/5964334/39182